adapters/Pangle/PangleAdapter/GADMediationAdapterPangle.m

```diff
@@ -29,13 +31,19 @@ @implementation GADMediationAdapterPangle {
   GADPangleRTBInterstitialRenderer *_interstitialRenderer;
   /// Pangle rewarded ad wrapper.
   GADPangleRTBRewardedRenderer *_rewardedRenderer;
+  /// Pangle native ad wrapper.
+  GADPangleRTBNativeRenderer *_nativeRenderer;
 }
 
 - (void)collectSignalsForRequestParameters:(nonnull GADRTBRequestParameters *)params
                          completionHandler:
-                             (nonnull GADRTBSignalCompletionHandler)completionHandler {
-  NSString *signals = [BUAdSDKManager getBiddingToken:nil];
-  completionHandler(signals, nil);
+(nonnull GADRTBSignalCompletionHandler)completionHandler {
+    GADPangleNetworkExtras *extras = [params.extras isKindOfClass:[GADPangleNetworkExtras class]] ? params.extras : nil;
+    if (extras && extras.userDataString.length > 0) {
+        [BUAdSDKConfiguration configuration].userExtData = extras.userDataString;
+    }
```

Comment:
@wangyuanyuanyy Since you resolved the conversation I previously added, I will create another one.

This doesn't look like the right place to pass the extras.

I assume this needs to be passed before Pangle SDK initialize:
https://github.com/googleads/googleads-mobile-ios-mediation/blob/master/adapters/Pangle/PangleAdapter/GADMediationAdapterPangle.m#L65